File: ios/Classes/MimsdkdemoPlugin.m
#import "MimsdkdemoPlugin.h"
#import "libMIMSDK_iOS.h"
@implementation MimsdkdemoPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"mimsdkdemo"
            binaryMessenger:[registrar messenger]];
  MimsdkdemoPlugin* instance = [[MimsdkdemoPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"getPlatformVersion" isEqualToString:call.method]) {

     //这里可以直接调用libMIMSDK_iOS 里的方法

    result([@"iOS " stringByAppendingString:[[UIDevice currentDevice] systemVersion]]);
  } else {
    result(FlutterMethodNotImplemented);
  }
}

@end

File: macos/Classes/MimsdkdemoPlugin.swift
import Cocoa
import FlutterMacOS

public class MimsdkdemoPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "mimsdkdemo", binaryMessenger: registrar.messenger)
    let instance = MimsdkdemoPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":

      result("macOS " + ProcessInfo.processInfo.operatingSystemVersionString)
    default:
      result(FlutterMethodNotImplemented)
    }
  }

  @_silgen_name("GetFriendList") // 将c函数c_add的imp作为被标注的方法的imp
  func swift_GetFriendList()   // 这个swift方法的imp实际上被指向了c函数c_add

}
